.github/workflows/windows.yml

```diff
@@ -74,3 +72,54 @@ jobs:
           Get-ChildItem env:
           cd $env:PYTHONPATH
           python -c "import sys; import scipy; sys.exit(not scipy.test())"
+
+
+  #############################################################################
+  full_dev_py_min_numpy:
+    name: cp39 (meson) full, dev.py, minimum numpy
+    if: "github.repository == 'scipy/scipy' || github.repository == ''"
+    runs-on: windows-2019
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.9'
+          cache: 'pip'
+          cache-dependency-path: 'environment.yml'
+
+      - name: Win_amd64 - install rtools
+        run: |
+          # mingw-w64
+          choco install rtools --no-progress
+          echo "c:\rtools40\ucrt64\bin;" >> $env:GITHUB_PATH
+
+      - name: Install OpenBlas
+        shell: bash
+        run: |
+          set -xe
+          bash tools/wheels/cibw_before_build_win.sh .
+          echo "PKG_CONFIG_PATH=c:\opt\64\lib\pkgconfig;" >> $GITHUB_ENV
+
+      - name: pip-packages
+        run: |
+          # 1.22.3 is currently the oldest numpy usable on cp3.9 according
+          # to pyproject.toml
+          python -m pip install numpy==1.22.3 cython pybind11 pythran meson-python meson ninja pytest pytest-xdist pytest-timeout pooch rich_click click doit pydevtool
+
+      - name: Build
+        run: |
+          $env:SCIPY_USE_PYTHRAN=1
+          python dev.py --build-dir=$pwd\build --install-prefix=$pwd\build-install build -j 2 --win-cp-openblas
+          cd build
+          meson install
+          cd ..
```

Comment:
I found that these lines are required. Without the lines I can't import e.g. `scipy.special` because none of the extension modules can be imported (i.e. ImportError can't import _ufuncs). This problem occurs locally as well. I'll comment the lines out to illustrate the issue.

Reply:
Hmm very odd. Maybe it's due to a bug in `meson-python` that was recently fixed (running `ninja` rather than `meson install`, it's back to `meson install` in 0.13.0).

In case it is needed for now, it can be `meson install -C build`, to avoid the `cd`'ing.

Reply:
This issue is illustrated in [5c34c9f](https://github.com/scipy/scipy/pull/18325/commits/5c34c9ffa152d6632fbc6e2ff412caf780156df1). Here we have a simple `python dev.py test` run, but the extension modules aren't found. The log is here:

https://github.com/scipy/scipy/actions/runs/4759148604/jobs/8458086451?pr=18325#step:8:32

Reply:
I think I found the problem, but I'll test with this new job on my own fork first tomorrow.